apps/token-manager/app/src/components/InfoBoxes.js

```diff
@@ -0,0 +1,209 @@
+import React from 'react'
+import styled from 'styled-components'
+import {
+  Box,
+  GU,
+  Text,
+  TokenBadge,
+  breakpoint,
+  textStyle,
+  theme,
+  useTheme,
+} from '@aragon/ui'
+import { useNetwork } from '@aragon/api-react'
+import { formatBalance, stakesPercentages } from '../utils'
+import You from './You'
+import LocalIdentityBadge from './LocalIdentityBadge/LocalIdentityBadge'
+
+const DISTRIBUTION_ITEMS_MAX = 7
+
+const displayedStakes = (accounts, total, colors) => {
+  return stakesPercentages(accounts.map(({ balance }) => balance), {
+    total,
+    maxIncluded: DISTRIBUTION_ITEMS_MAX,
+  }).map((stake, index) => ({
+    name: stake.index === -1 ? 'Rest' : accounts[index].address,
+    stake: stake.percentage,
+    color: colors[index % colors.length],
+  }))
+}
+
+class InfoBoxes extends React.PureComponent {
+  static defaultProps = {
+    holders: [],
+  }
+  transferableLabel() {
+    const { tokenTransfersEnabled } = this.props
+    if (tokenTransfersEnabled === undefined) {
+      return 'Unknown'
+    }
+    return tokenTransfersEnabled ? 'Yes' : 'No'
+  }
+  render() {
+    const {
+      holders,
+      network,
+      tokenAddress,
+      tokenDecimalsBase,
+      tokenName,
+      tokenSupply,
+      tokenSymbol,
+      userAccount,
+      theme,
+      ...rest
+    } = this.props
+
+    const colors = [
+      theme.blue,
+      theme.red,
+      theme.brown,
+      theme.yellow,
+      theme.purple,
+      theme.green,
+    ]
+
+    const stakes = displayedStakes(holders, tokenSupply, colors)
+    return (
+      <React.Fragment>
+        <Box heading="Token Info">
+          <ul
+            css={`
+              color: ${theme.surfaceContent};
+            `}
+          >
+            {[
+              [
+                'Total supply',
+                <strong>
```

Comment:
Should this be a `textStyle`?

Reply:
Maybe yes, we could have stylistic variants for some of the text styles, like `body2-bold`, which could be useful to have directly. Let’s discuss it with the design team.

For the `strong` element itself, I think we should keep it as it also carries a semantic value.